t/op/chr.t

```diff
@@ -19,67 +19,112 @@ is(chr(127), "\x7F");
 is(chr(128), "\x80");
 is(chr(255), "\xFF");
 
-is(chr(-0.1), "\x{FFFD}"); # The U+FFFD Unicode replacement character.
-is(chr(-1  ), "\x{FFFD}");
-is(chr(-2  ), "\x{FFFD}");
-is(chr(-3.0), "\x{FFFD}");
+{
+    my @these_warnings = ();
+    local $SIG{__WARN__} = sub { push @these_warnings, $_[0]; };
+    for my $arg ( -0.1, -1, -2, -3.0 ) {
+        is(
+            chr($arg),
+            "\x{FFFD}",
+            "For '$arg' got Unicode replacement character (U+FFFD)"
+        );
+    }
+    my $expected = 4;
+    is(
+        @these_warnings,
+        $expected,
+        "Got $expected 'invalid negative number' warnings, as expected"
+    );
```

Comment:
a better choice would have been to check the warning state after each call to `chr` then reset the array

Reply:
What I propose is that in the **first** of the three blocks where we're testing `chr` on negative numbers, we test for both the count of warnings and for a match on the warning message itself.

Then in the 2 other blocks farther down (the first of which you reference below), we leave things as they are, i.e., we simply test for the count of warnings collected.  We will have already demonstrated that we can test for the content of the warning message in the first block.  Please look at next commit.